analyses/cnv-frequencies/01-cnv-frequencies.py

```diff
@@ -0,0 +1,249 @@
+"""
+
+
+__author__ = ('Eric Wafula (wafulae@chop.edu)')
+__version__ = '1.0'
+__date__ = '12 July 2021'
+
+
+import os 
+import sys
+import csv
+import json
+import mygene
+import argparse
+import numpy as np
+import pandas as pd
+from functools import reduce
+
+
+def read_parameters():
+     p = argparse.ArgumentParser(description=("The 01-snv-frequencies.py scripts creates copy number variation (CNV) cancer type and"      
+                                             " study gene-level alterations frequencies table for the OPenPedCan analyses modules."),
+                                formatter_class=argparse.RawTextHelpFormatter)
+     p.add_argument('HISTOLOGY_FILE', type=str, default=None,
+                    help="OPenPedCan histology file (histologies.tsv)\n\n")
+     p.add_argument('CNV_FILE', type=str, default=None,
+                    help="OPenPedCan CNV consensus file (consensus_seg_annotated_cn_autosomes.tsv or consensus_seg_annotated_cn_x_and_y.tsv)\n\n")
+     p.add_argument('PRIMARY_TUMORS', type=str, default=None,
+                    help="OPenPedCan independent primary tumor samples file (independent-specimens.wgs.primary.tsv)\n\n")
+     p.add_argument('RELAPSE_TUMORS', type=str, default=None,
+                    help="OPenPedCan independent relapse tumor samples file (independent-specimens.wgs.relapse.tsv)\n\n")
+     p.add_argument('EFO_MONDO', type=str, default=None,
+                    help="OPenPedCan disease to EFO/MONDO mapping file (efo-mondo-map.tsv)\n\n")
+     p.add_argument('ENSG_RMTL', type=str, default=None,
+                    help="OPenPedCan Ensembl to RMTL mapping file (ensg-hugo-rmtl-v*-mapping.tsv)\n\n")
+     p.add_argument('-v', '--version', action='version',
+                   version=f"classify_reads.py version {__version__} ({__date__})",
+                   help="Print the current 01-snv-frequencies.py version and exit\n\n")
+     return p.parse_args()
+
+
+def merge_histology_and_cnv_data(histology_file, cnv_consensus_file):
+	# load histology file
+	histology_df = pd.read_csv(histology_file, sep="\t", na_filter=False, dtype=str)
+	histology_df.head()
+
+	# subset histology dataframe for relevant columns
+	histology_df = histology_df[["Kids_First_Biospecimen_ID","Kids_First_Participant_ID", 
+	                                    "cohort", "cancer_group", "sample_type"]]
+
+	# load CNV consensus file
+	cnv_df = pd.read_csv(cnv_consensus_file, sep="\t")
+	#cnv_df.head()
+
+	# merge subset of histology dataframe to CNV dataframe keeping only sample
+	# present in the CNV table (left outer join)
+	merged_df = pd.merge(cnv_df, histology_df, how="left", left_on="biospecimen_id", 
+	                     right_on="Kids_First_Biospecimen_ID")
+
+	# check if non tumor sample are present in the merged dataframe 
+	if  not np.array_equal(np.array(["Tumor"]), merged_df.sample_type.unique()):
+	    raise Exception(f"Merged hsitology-CNV dataframe contains non tumor samples")
```

Comment:
```suggestion
	    raise Exception("Merged hsitology-CNV dataframe contains non tumor samples")
```

Literal string interpolation (`f"Merged ..."`) is not supported by the python3.5 version in the docker image.